src/main/java/org/folio/innreach/batch/contribution/service/ContributionJobRunner.java

```diff
@@ -305,24 +364,47 @@ private void contributeItemsChunk(UUID centralServerId, String bibId, List<Item>
       stats.addRecordsTotal(itemsCount);
       recordContributionService.contributeItems(centralServerId, bibId, items);
       stats.addRecordsContributed(itemsCount);
-    } catch (Exception e) {
+      ContributionJobRunner.recordsProcessed.put(getContributionJobContext().getTenantId(), recordsProcessed.get(getContributionJobContext().getTenantId()) == null ? 1
```

Comment:
What is stopping ongoing contributions from incrementing the `recordsProcessed` counter? Because of course if that happens the count will be wrong.

Reply:
@Arghya22 is it that the `ThreadLocal` of the `ContributionJobContextManager` handles this? cc @gurleenkaurbp

Reply:
should we do **onGoing** contribution ( updating /creating marc records ) when initialContribution is running ? Why someone wants to change it when they ( tenant ) already triggered initial Contribution !

Reply:
Unfortunately it looks like stopping ongoing contributions isn't an option. We can't ask them to stop things like performing checkouts and such. So we need to fix this ASAP and get a new build to Carole.

Reply:
checks are added to avoid incrementation of count for ongoing contribution. But having said this , ongoing contrbution retry fix is not part of this jira/PR , so testing should be done only for **Initial contribution** and ongoing contribution ( eg. update / add MARC instance,holding or item should be avoided while testing this PR ) . All fix of ongoing contrbution will be part of **[MODINREACH-351](https://issues.folio.org/browse/MODINREACH-351)**